Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -232,8 +234,9 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
             }
             let pbxBuildFile = PBXBuildFile(file: fileReference, settings: settings)
             pbxproj.add(object: pbxBuildFile)
-            headersBuildPhase.files?.append(pbxBuildFile)
+            pbxBuildFiles.append(pbxBuildFile)
         }
+        headersBuildPhase.files = pbxBuildFiles
```

Comment:
Sadly there's a regression here where headers get omitted from the generated project :/

the closure above is invoked by the lines below

```swift
try headers.private.sorted().forEach { try addHeader($0, "private") }
try headers.public.sorted().forEach { try addHeader($0, "public") }
try headers.project.sorted().forEach { try addHeader($0, nil) }
```

as such by the point here where we are assigning the files, it's empty. 

A possible fix could be to make the closure return the build files which we could add to the headers build phase and pbxproj at the very end or assigning the header files after calling the closures.

Reply:
Oops, yes, we would just need to move it after the hearders, my bad, thanks for catching it!